```
Cells were assigned to cell groups using Monocle3 with UMAP clustering [1-11]. Data was normalized to remove batch effects using PCA clustering to 100 dimensions. Pseudotime was calculated in Monocle3, with colors representing pseudotime changes among the cell clusters. The top 3 marker genes expressed in each cluster were plotted with at least 10% of cells expressing the respective genes, using 10 cells as reference.  Bioinformatic methods for Monocle analyses can be found at: https://github.com/ISUgenomics/SingleCellRNAseq_RyanSmith.

1.	Trapnell, C., et al., The dynamics and regulators of cell fate decisions are revealed by pseudotemporal ordering of single cells. Nature biotechnology, 2014. 32(4): p. 381.
2.	Cao, J., et al., Comprehensive single-cell transcriptional profiling of a multicellular organism. Science, 2017. 357(6352): p. 661-667.
3.	Levine, J.H., et al., Data-driven phenotypic dissection of AML reveals progenitor-like cells that correlate with prognosis. Cell, 2015. 162(1): p. 184-197.
4.	Wolf, F.A., et al., PAGA: graph abstraction reconciles clustering with trajectory inference through a topology preserving map of single cells. Genome biology, 2019. 20(1): p. 59.
5.	Traag, V.A., L. Waltman, and N.J. van Eck, From Louvain to Leiden: guaranteeing well-connected communities. Scientific reports, 2019. 9(1): p. 1-12.
6.	Haghverdi, L., et al., Batch effects in single-cell RNA-sequencing data are corrected by matching mutual nearest neighbors. Nature biotechnology, 2018. 36(5): p. 421-427.
7.	Cao, J., et al., The single-cell transcriptional landscape of mammalian organogenesis. Nature, 2019. 566(7745): p. 496-502.
8.	Packer, J.S., et al., A lineage-resolved molecular atlas of C. elegans embryogenesis at single-cell resolution. Science, 2019. 365(6459): p. eaax1971.
9.	Sharon, N., et al., A peninsular structure coordinates asynchronous differentiation with morphogenesis to generate pancreatic islets. Cell, 2019. 176(4): p. 790-804. e13.
10.	Qiu, X., et al., Reversed graph embedding resolves complex single-cell trajectories. Nature methods, 2017. 14(10): p. 979.
11.	Qiu, X., et al., Single-cell mRNA quantification and differential analysis with Census. Nature methods, 2017. 14(3): p. 309.

```
